### PR TITLE
Refactor-out `parse_config_file()`

### PR DESCRIPTION
**Pull request by atugushev**
Refs: https://github.com/jazzband/pip-tools/pull/1913

Make it more clean

### CODE REVIEW

File: piptools/utils.py [hunk 1]
```diff
@@ -696,35 +699,48 @@ def parse_config_file(
...
+
+def _invert_negative_bool_options(
+    ctx: click.Context, config: dict[str, Any]
+) -> dict[str, Any]:
+    new_config = {}
+    cli_opts = get_cli_options(ctx)
+    for key, value in config.items():
+        long_option = "--" + key
+        option_name = cli_opts[long_option].name
+        assert option_name is not None
```
Comment (chrysle) [inline piptools/utils.py]:
```suggestion
        option_name = cli_opts.get(long_option, "invalid")
        if option_name is not "invalid":
            option_name = option_name.name
```

Comment (atugushev) [reply piptools/utils.py]: Fixed in 4ee4de3ba11a41f86b70bedb1c41d2fdce1ab6e4

**Comment by atugushev** [comment]
I'm still not happy with this code, I think it can be improved. But let's not block on this now and refactor this in a follow-up PR.

**Comment by chrysle** [comment]
Thanks for the changes!

**Pull request merged**